hicommonwealth/commonwealth
# User Sign In/Onboarding Fixes

## Link to Issue
Closes: #8653 
Closes: #8645

## Description of Changes
- Adds a migration which updates `User.is_welcome_onboard_flow_complete`  to true if the user already has a profile name or user id set.
- Modifies `getAddressProfile` to work with empty community id arrays
- Modifies client authentication flow to ignore current or default chain when signing in.
  - This is because a chain is not required to find a specific user via address.

## Test Plan
1. Go to http://localhost:8080/vita-fast/ and create an account with a _new_ MetaMask address
2. Sign out
3. Go to http://localhost:8080/dashboard/global
4. Sign in with the same address you just created an account with
5. Sign out
6. Go to http://localhost:8080/dydx
7. Sign in with the same address
8. Join the dydx community

At no point after step 1 should it ask you to created a new account or re-complete the new onboarding flow.

To test the migration for existing users, create a new wallet address and execute the following query:
```
UPDATE "Addresses"
SET address = '[paste your wallet address]'
WHERE address = '0xFcA51ed3FdDe1c66Ed5c37f5C048cCBc9049Ed66';
```
Then attempt to sign in. You should not be prompted with an onboarding flow (the query mocks user from #8645)

## Deployment Plan

1. 

## Other Considerations

- Must be merged after #8647
